### DISCUSSION TITLE
How to deploy the flow bite Hugo website ?

### DISCUSSION THREAD

**Discussion by sid-r-singh** [themesberg/flowbite#29] [Q&A] [answered]
Hi there! I found no instructions in the README of the [flowbite repo](https://github.com/themesberg/flowbite) regarding developing/deploying the site itself (using Hugo). It will be nice if you can mention all the commands that one is supposed to run in order, in order to preview (or deploy) this Hugo website.

**Comment by zoltanszogyenyi** [comment, marked as answer]
Hey @sid-r-singh,

You should first set the `baseURL` from `config.yml` and then run `npm run build` and you'll find the files inside the `gh_pages/` folder. We'll create a guide of working with the documentation itself with the release of v1.2.1.

**Reply by sid-r-singh** [reply]
Thanks for the reply. Looking forward to the guide for working with the documentation. Please also do mention the instructions for setting up the development environment & commands for deploying the site.

**Comment by PwrSrg** [comment]
I **_think_** I have a very similar question, but not necessarily about Hugo.

I get the basic concept of grabbing the HTML snippets from the components site, and dropping them into your code.  But I am not quite grasping the rest of the process.  The underlying issue is that I am having a hard time wrapping my head around how Flowbite actually **WORKS**, and unfortunately the "docs" are no help.

From what I understand, Flowbite is simply a collection of pre-built (HTML) components assembled using the default Tailwind utility classes, same as **Tailwind UI**.  But if that were the case, then WHY is there an "install" at all?  What is NPM actually "installing"?  And WHY does it install a `flowbite.css`?  Shouldn't there be nothing more than a neatly organized collection of HTML snippets to use in your projects, along with a JS file to make the interactive elements function?  And once I "install" Flowbite via NPM, WHERE are all of the components??  I can't seem to find them anywhere.  For example - the Card components contain the text "Noteworthy technology", but a full project text search returns 0 results.

**What am I missing?? 🤔**

A **video** or **tutorial** showing the entire process from start to finish would be ***extremely*** helpful, including a detailed explanation of the Flowbite methodology and workflow.  So would complete docs. 🙃  Or at least a simple code snippet at the end of Getting Started to _verify_ your installation by loading a sample component in the browser.  A "Next Steps" section would be even nicer!  Because right now I have no idea what to do next, what to look for, or if everything is even "installed correctly" or not.  The docs just feel half-baked, and kind of leave you hanging to fend for yourself. 😟

Sorry for so many questions!  I really hope I am not the _only_ one wondering these things.
I also wasn't sure where to post this, and Reditt would have turned into a **FRENZY**!!  😆

**Comment by zoltanszogyenyi** [comment]
Hey @PwrSrg,

Thanks for the comment and the feedback.

The plugin is used to apply some styles to pseudo elements like the forms so the HTML doesn't get too bloated. It is also used to apply some extra utility classes on top of the default ones provided by Tailwind CSS.

The `flowbite.css` file is only generated for the CDN feature that we recommend only using it when testing out the library, but not on production. It gets published to NPM so it is available via `https://unpkg.com/`.

I'll review the onboarding flow on the documentation based on your recommendations.

**Reply by PwrSrg** [reply]
Thank you for the info.  I will keep playing with it, and look forward to the documentation updates.

**Reply by mirastalbayev** [reply]
Hello everyone!! Can you tell me if the documentation has been updated? Since I am facing many problems. Components get small, icons get too big, etc. It seems like I installed everything correctly.
